lib/smart_proxy_openscap/foreman_forwarder.rb

```diff
@@ -9,14 +11,13 @@ def post_arf_report(cname, policy_id, date, data)
         response = send_request(foreman_api_path, json)
         # Raise an HTTP error if the response is not 2xx (success).
         response.value
-        res = JSON.parse(response.body)
-        raise StandardError, "Received response: #{response.code} #{response.msg}" unless res['result'] == 'OK'
-      rescue StandardError => e
-        logger.debug response.body if response
-        logger.debug e.backtrace.join("\n\t")
+        JSON.parse(response.body)
+      rescue Net::HTTPServerException => e
+        logger.debug "Received response: #{response.code} #{response.msg}"
+        logger.debug response.body
+        raise ReportUploadError, e.message if response.code.to_i == 422
```

Comment:
It would probably be safer to raise this when response code is not 200.

Reply:
The exception does get re-raised, see the next line. I just need to distinguish the 422 case and handle it separately in `openscap_api`.